src/components/Reports/GoalCalculator/CalculatorSettings/Categories/SettingsCategory/SettingsCategory.tsx

```diff
@@ -0,0 +1,88 @@
+import { useRouter } from 'next/router';
+import React, { useRef } from 'react';
+import { TextField } from '@mui/material';
+import { Field, Form, Formik } from 'formik';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { useDebouncedCallback } from 'src/hooks/useDebounce';
+import { getQueryParam } from 'src/utils/queryParam';
+import { useGoalCalculationQuery } from '../../../Shared/GoalCalculation.generated';
+import { useUpdateGoalCalculationMutation } from './GoalCalculation.generated';
+
+export const SettingsCategory: React.FC = () => {
+  const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+  const { query } = useRouter();
+  const goalCalculationId = getQueryParam(query, 'goalCalculationId') ?? '';
+  const hasUserInteracted = useRef(false);
+
+  const shouldFetch = !!accountListId && !!goalCalculationId;
+  const { data: goalData } = useGoalCalculationQuery({
+    variables: { accountListId, id: goalCalculationId },
+    skip: !shouldFetch,
+  });
+
+  const [updateGoalCalculationMutation] = useUpdateGoalCalculationMutation();
+
+  const debouncedUpdateGoalTitle = useDebouncedCallback((title: string) => {
+    updateGoalCalculationMutation({
```

Comment:
Can you wrap the mutation in `trackMutation` from `useGoalCalculator()`? That will make the loading spinner spin while it is saving.